In [8]:
import seaborn as sns
titanic = sns.load_dataset("titanic")

df = titanic.copy()


In [6]:
df.shape

(891, 15)

il ya 891 lignes est 15 colonnes

In [7]:
df.dtypes

survived          int64
pclass            int64
sex                 str
age             float64
sibsp             int64
parch             int64
fare            float64
embarked            str
class          category
who                 str
adult_male         bool
deck           category
embark_town         str
alive               str
alone              bool
dtype: object

voici les types de colonnes

In [9]:
df.head()

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


on constate que les colonnes:
"survived" et "alive"
"pclass" et "class"
"sex" et "who"
"embarked" et "embark_town"


In [10]:
df.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

In [11]:
df["class"].value_counts()

class
Third     491
First     216
Second    184
Name: count, dtype: int64

In [12]:
df["embark_town"].value_counts()

embark_town
Southampton    644
Cherbourg      168
Queenstown      77
Name: count, dtype: int64

La troisième classe est de loin la plus représentée, concentrant plus de la moitié des passagers du navire.   
Southampton est le point de départ principal, totalisant une majorité écrasante des embarquements par rapport à Cherbourg et Queenstown.  
La variable deck souffre d'un manque critique de données avec 688 valeurs manquantes, ce qui la rend inexploitable en l'état